app/components/canvas/canvas.js

```diff
@@ -78,6 +78,52 @@ class CanvasController {
 
   setupMouseEventHandlers_() {
     this.canvas_
+        .on('mousedown', event => {
```

Comment:
i would refactor this into a function like `hitTest_(x, y)` that returns a layer inside `this.artwork`